File: unity/Assets/Scripts/Multiplayer/Matchmaking/Requests/MatchmakingRequestStatusResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking.Requests
{
    /// <summary>
    /// Response containing the status of a specific join request.
    /// </summary>
    [System.Serializable]
    public class MatchmakingRequestStatusResponse : ApiResponse<EMatchmakingStatusError>
    {
        /// <summary>
        /// The join request information including status and details.
        /// </summary>
        public MatchmakingRequestInfo request;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/requests/MatchmakingRequestStatusResponse.java
package com.michitai.multiplayer.matchmaking.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response containing the status of a specific join request.
 */
public class MatchmakingRequestStatusResponse extends ApiResponse {
    @JsonProperty("request")
    private MatchmakingRequestInfo request;

    public MatchmakingRequestInfo getRequest() {
        return request;
    }

    public void setRequest(MatchmakingRequestInfo request) {
        this.request = request;
    }
}
